tests/test_jsvalidation.py

```diff
@@ -0,0 +1,597 @@
+    obj = JSONSchemaValidator()
+    obj.validate(v1, 'first')
+
+
+def test_root_not_object():
+    """Test that not having type 'object' of the root fails."""
+    v1 = {
+        "fieldA": {
+            "type": "string"
+        }
+    }
+    obj = JSONSchemaValidator()
+    with pytest.raises(JSONSchemaCompatibilityError):
+        obj.validate(v1, 'first')
+
+
+def test_dependencies_not_dict():
+    """Test that having a conflict when dependencies are neither of type list
+    nor a dict."""
+    v1 = {
+        "type": "object",
+
+        "properties": {
+            "name": {"type": "string"},
+            "credit_card": {"type": "number"},
+        },
+
+        "required": ["name"],
+
+        "dependencies": {
+            "name": "credit_card"
+        }
+    }
+    obj = JSONSchemaValidator()
+    with pytest.raises(ValueError):
+        obj.validate(v1, 'first')
+
+
+def test_conflict_in_dependencies():
+    """Test that having conflict between elements in dependencies fails."""
+    v1 = {
+        "type": "object",
+
+        "properties": {
+            "name": {"type": "string"},
+            "credit_card": {"type": "number"},
+            "billing_address": {"type": "number"}
+        },
+
+        "required": ["name"],
+
+        "dependencies": {
+            "credit_card": {
+                "properties": {
+                    "billing_address": {"type": "string"}
+                },
+                "required": ["billing_address"]
+            }
+        }
+    }
+    obj = JSONSchemaValidator()
+    with pytest.raises(JSONSchemaCompatibilityError):
+        obj.validate(v1, 'first')
+
+
+def test_no_confict_in_collection_pass():
+    """Test that having no conflict between elements of a collection passes."""
+    v1 = {
+        "type": "object",
+        "oneOf": [
+            {
+                "type": "object",
+                "properties": {
+                    "field_A": {"type": "string"}
+                }
+            }, {
+                "type": "object",
+                "properties": {
+                    "field_B": {"type": "integer"}
+                }
+            }
+        ]
+    }
+    obj = JSONSchemaValidator()
+    obj.validate(v1, 'first')
+
+
+def test_conflict_in_collection():
+    """Test that having a conflict between elements of a collection fails."""
+    v1 = {
+        "type": "object",
+        "oneOf": [
+            {
+                "type": "object",
+                "properties": {
+                    "field_A": {"type": "string"}
+                }
+            }, {
+                "type": "object",
+                "properties": {
+                    "field_A": {"type": "integer"}
+                }
+            }
+        ]
+    }
+    obj = JSONSchemaValidator()
+    with pytest.raises(JSONSchemaCompatibilityError):
+        obj.validate(v1, 'first')
+
+
+def test_conflict_in_collection2():
+    """Test that having a conflict between elements of a collection fails."""
```

Comment:
As the previous test and this one are very similar it is good to describe the difference between them. That way the next person reading the documentation will not have to search for the difference in the code.

We can replace in this docstring:
"""...between **properties of an allOf** collection..."""

And in the docstring of the previous test:

"""...between **properties of a oneOf** collection..."""